### ISSUE TITLE
Nuget on OS X works only with linux path slashes (/) not the default windows (\)

### ISSUE THREAD

**Issue by pfaucon** [NuGet/Home#5914]
## Details about Problem

NuGet product used (NuGet.exe | VS UI | Package Manager Console | dotnet.exe):
NuGet.exe on mac OS X, called from the terminal

NuGet version (x.x.x.xxx):
2.12.0.0

OS version (i.e. win10 v1607 (14393.321)):
OS X Sierra 10.12.6

Worked before? If so, with which NuGet version:
Not sure

## Detailed repro steps so we can see the same problem

1. create basic nuspec file:

```
echo '<?xml version="1.0"?>
<package >
  <metadata>
    <id>test.nuspec</id>
    <version>1.0.0</version>
    <title>try forward slashes</title>
    <authors>Christophe Faucon</authors>
    <owners>Christophe Faucon</owners>
    <projectUrl>https://fakeurl.com</projectUrl>
    <requireLicenseAcceptance>false</requireLicenseAcceptance>
    <description>try forward slashes</description>
  </metadata>
  <files>
    <file src="..\test.txt" target="lib\MonoAndroid10\test.txt" />
  </files>
</package>' > test.nuspec

```

2. create text file 
`echo 'hello!' > ../test.txt`

3. nuget pack -NoDefaultExcludes test.nuspec
(It looks like #913 is still a bug, despite having been closed)

4. modify the <file> src attribute to use forward slash instead and everything works fine

### Verbose Logs

nuget pack -NoDefaultExcludes -verbosity detailed test.nuspec
Attempting to build package from 'test.nuspec'.
System.IO.DirectoryNotFoundException: Could not find a part of the path '/Users/philippefaucon/Documents/Development/Mobil.ISY.Forms/Mobil.ISY.Forms/nuspec/..\'.
  at System.IO.__Error.WinIOError (System.Int32 errorCode, System.String maybeFullPath) [0x000f7] in <48b95f3df5804531818f80e28ec60191>:0 
  at System.IO.FileSystemEnumerableIterator`1[TSource].HandleError (System.Int32 hr, System.String path) [0x00006] in <48b95f3df5804531818f80e28ec60191>:0 
  at System.IO.FileSystemEnumerableIterator`1[TSource].CommonInit () [0x00054] in <48b95f3df5804531818f80e28ec60191>:0 
  at System.IO.FileSystemEnumerableIterator`1[TSource]..ctor (System.String path, System.String originalUserPath, System.String searchPattern, System.IO.SearchOption searchOption, System.IO.SearchResultHandler`1[TSource] resultHandler, System.Boolean checkHost) [0x000d6] in <48b95f3df5804531818f80e28ec60191>:0 
  at System.IO.FileSystemEnumerableFactory.CreateFileNameIterator (System.String path, System.String originalUserPath, System.String searchPattern, System.Boolean includeFiles, System.Boolean includeDirs, System.IO.SearchOption searchOption, System.Boolean checkHost) [0x00009] in <48b95f3df5804531818f80e28ec60191>:0 
  at System.IO.Directory.InternalGetFileDirectoryNames (System.String path, System.String userPathOriginal, System.String searchPattern, System.Boolean includeFiles, System.Boolean includeDirs, System.IO.SearchOption searchOption, System.Boolean checkHost) [0x00000] in <48b95f3df5804531818f80e28ec60191>:0 
  at System.IO.Directory.InternalGetFiles (System.String path, System.String searchPattern, System.IO.SearchOption searchOption) [0x00000] in <48b95f3df5804531818f80e28ec60191>:0 
  at System.IO.Directory.GetFiles (System.String path, System.String searchPattern, System.IO.SearchOption searchOption) [0x00038] in <48b95f3df5804531818f80e28ec60191>:0 
  at NuGet.PathResolver.PerformWildcardSearchInternal (System.String basePath, System.String searchPath, System.Boolean includeEmptyDirectories, System.String& normalizedBasePath) [0x00080] in <05c5fafa88254577a971f1c705dfdb56>:0 
  at NuGet.PathResolver.ResolveSearchPattern (System.String basePath, System.String searchPath, System.String targetPath, System.Boolean includeEmptyDirectories) [0x00014] in <05c5fafa88254577a971f1c705dfdb56>:0 
  at NuGet.PackageBuilder.AddFiles (System.String basePath, System.String source, System.String destination, System.String exclude) [0x00000] in <05c5fafa88254577a971f1c705dfdb56>:0 
  at NuGet.PackageBuilder.PopulateFiles (System.String basePath, System.Collections.Generic.IEnumerable`1[T] files) [0x00024] in <05c5fafa88254577a971f1c705dfdb56>:0 
  at NuGet.PackageBuilder.ReadManifest (System.IO.Stream stream, System.String basePath, NuGet.IPropertyProvider propertyProvider) [0x00037] in <05c5fafa88254577a971f1c705dfdb56>:0 
  at NuGet.PackageBuilder..ctor (System.String path, System.String basePath, NuGet.IPropertyProvider propertyProvider, System.Boolean includeEmptyDirectories) [0x0000f] in <05c5fafa88254577a971f1c705dfdb56>:0 
  at NuGet.Commands.PackCommand.CreatePackageBuilderFromNuspec (System.String path) [0x0005b] in <5d68591887e84a69ba8a434533c2d2d6>:0 
  at NuGet.Commands.PackCommand.BuildFromNuspec (System.String path) [0x00000] in <5d68591887e84a69ba8a434533c2d2d6>:0 
  at NuGet.Commands.PackCommand.BuildPackage (System.String path) [0x00013] in <5d68591887e84a69ba8a434533c2d2d6>:0 
  at NuGet.Commands.PackCommand.ExecuteCommand () [0x000a5] in <5d68591887e84a69ba8a434533c2d2d6>:0 
  at NuGet.Commands.Command.Execute () [0x000ca] in <5d68591887e84a69ba8a434533c2d2d6>:0 
  at NuGet.Program.Main (System.String[] args) [0x0018b] in <5d68591887e84a69ba8a434533c2d2d6>:0

**Comment by emgarten** [comment]
> NuGet version (x.x.x.xxx):
2.12.0.0

Use the latest 4.3.0 nuget.exe from: https://www.nuget.org/downloads

2.x did not officially support cross platform scenarios, and 2.12 nuget.exe is not an official release as far as I know.

**Comment by rohit21agrawal** [comment]
how is nuget.exe being used on mac? are you using it via mono?

**Comment by pfaucon** [comment]
It was automatically installed with Visual Studio for mac, but yes through Mono. It looks like something is awry with the VS installer though.  Inside of VS it was already using NuGet 4.3, but from the command line "nuget" maps to a script:
/Library/Frameworks/Mono.framework/Versions/Current/Commands/nuget
```
#!/bin/sh
exec /Library/Frameworks/Mono.framework/Versions/5.2.0/bin/mono $MONO_OPTIONS /Library/Frameworks/Mono.framework/Versions/5.2.0/lib/mono/nuget/NuGet.exe "$@"
```

But that specific version of NuGet.exe was the 2.12.0.0 version.  I downloaded the new version and just replaced it and everything seems fine.  I now get [this](https://github.com/NuGet/Home/issues/5019) warning, but I'm not worried about it.

Thank you for the quick help though, it's greatly appreciated!

**Comment by emgarten** [comment]
I wasn't aware that nuget.exe shipped with VS for mac.

**Comment by pfaucon** [comment]
Or that 2 ship :) I think it must be the case though because this is a new mac for development and I haven't installed previous software (like mono directly, or the older Xamarin Studio).

**Comment by mrward** [comment]
NuGet.exe does not ship with VS Mac. VS Mac ships its own set of NuGet binaries, but not NuGet.exe. NuGet.exe is shipped with Mono. The latest Mono 5.4 ships with NuGet 4.3. Older Mono versions shipped with NuGet 2.12.

**Comment by pfaucon** [comment]
Interesting, I'm not sure where I picked up the old binary then, particularly since this mac was only set up a few weeks ago. 

I'm not totally convinced though, my chrome history shows that I reached a page "Thank you for downloading Visual Studio for Mac" 3 minutes before the filesystem shows those links appearing (11 days ago).  My assumption is that VS for Mac install mono (of which I am confident, since we are using it to build Xamarin, and historically Xamarin Studio used mac if I remember correctly), and for some reason the mono that is packed uses an older nuget.exe internally, though I don't know why that would be the case (maybe it ships with an old mono?).
